thirdeye-ui/src/app/components/impact-dashboard/summary/use-summary-data.tsx

```diff
@@ -77,8 +78,53 @@ export const useSummaryData = ({
                     name: "",
                 },
             },
+            notificationChannelsUsed: [],
         });
 
+    useEffect(() => {
+        const allAlertIds = anomalies?.map((anomaly) => anomaly.alert.id);
+        const allChannels: VerboseSummary["notificationChannelsUsed"] = [];
+        subscriptionGroups?.forEach((subscriptionGroup) => {
+            const assoaciatedAlerts = subscriptionGroup.alertAssociations?.map(
+                (alertAssociation) => alertAssociation.alert.id
+            );
+            if (
+                assoaciatedAlerts?.some((assoaciatedAlert) =>
+                    allAlertIds?.includes(assoaciatedAlert)
+                )
+            ) {
+                allChannels.push(
+                    ...subscriptionGroup.specs.map((spec) => spec.type)
+                );
+            }
+        });
+        if (!isEmpty(allChannels)) {
```

Comment:
nit: better to move out the `allChannels` computation outside of `useEffect` (probably `useMemo`) and only leave the `setVerbose...` part inside effect.

Reply:
the computation for channel is not heavy. So I believe useMemo would be on overkill here as with useMemo it will have to keep a reference of the dependencies and old computed value also in the memory and then based on the computation it will trigger the recalculation and then the useEffect will get triggered based on the channel.
Hence kept it in the one block only